docs/interfaces/interfaces-intro.md

```diff
@@ -0,0 +1,50 @@
+# Interfaces
+
+## Prerequisites
+
+* [Anatomy of an SDK Application](../basics/app-anatomy.md)
+* [Lifecycle of a Transaction](../basics/tx-lifecycle.md)
+
+
+## Synopsis
+
+Every application must include some interface users can use to interact with the defined functionalities. This document introduces user interfaces for SDK applications.
+
+- [Types of Application Interfaces](#types-of-application-interfaces)
+- [Module vs Application Interfaces](#module-vs-application-interfaces)
+  + [Module Developer Responsibilities](#module-developer-responsibilities)
+  + [Application Developer Responsibilities](#application-developer-responsibilities)
+
+
+## Types of Application Interfaces
+
+SDK applications should have a Command-Line Interface (CLI) and REST Interface to support HTTP requests. The SDK is opinionated about how to create these two interfaces; all modules specify [Cobra commands](https://github.com/spf13/cobra) and register routes using [Gorilla Mux routers](https://github.com/gorilla/mux). The CLI and REST Interface are conventionally defined in the application `/cmd/cli` folder.
+
+
+## Module vs Application Interfaces
+
+The process of creating an application interface is very different from creating a module interface, though the components are closely intertwined. As expected, the module interface handles the bulk of the underlying logic, unpacking user requests into arguments and routes, and neatly marshaling everything into ABCI requests to be handled by Baseapp. On the other hand, the application interface handles the user configurations and customizations, instantiates the application-specific values and objects, and passes everything to module interface functions.
```

Comment:
Application-level interfaces:

- Handles root (independent) level stuff like broadcasting, signing, status, etc.
- Aggregate module-level interfaces in order to route messages/queries to the appropriate module

Module-level interfaces:
- Let end-users create messages that will be handled by the module, and queries to the subset of the state defined by the module